File: stars-app/backend/models/campaign.py
from pydantic import BaseModel


class CampaignRow(BaseModel):
    campaign_name: str
    measure_code: str
    channel: str
    member_count: int
    projected_closures: int
    actual_closures: int | None = None
    lift_pct: str
    cost_str: str
    roi_str: str
    status: str


class ROISummary(BaseModel):
    total_members: int
    total_closures: int
    avg_roi: str
    total_cost: str


class CampaignResponse(BaseModel):
    items: list[CampaignRow]
    summary: ROISummary

File: stars-app/frontend/src/types/campaign.ts
export interface CampaignRow {
  campaign_name: string;
  measure_code: string;
  channel: string;
  member_count: number;
  projected_closures: number;
  actual_closures: number | null;
  lift_pct: string;
  cost_str: string;
  roi_str: string;
  status: string;
}

export interface ROISummary {
  total_members: number;
  total_closures: number;
  avg_roi: string;
  total_cost: string;
}

export interface CampaignResponse {
  items: CampaignRow[];
  summary: ROISummary;
}
